src/internal/common/idname.go

```diff
@@ -1,9 +1,47 @@
 package common
 
+import "golang.org/x/exp/maps"
+
 type IDNamer interface {
 	// the canonical id of the thing, generated and usable
 	//  by whichever system has ownership of it.
 	ID() string
 	// the human-readable name of the thing.
 	Name() string
 }
+
+type IDNameSwapper interface {
+	IDOf(name string) (string, bool)
+	NameOf(id string) (string, bool)
+	IDs() []string
+	Names() []string
+}
+
+var _ IDNameSwapper = &IDsNames{}
+
+type IDsNames struct {
+	IDToName map[string]string
+	NameToID map[string]string
```

Comment:
please don't export these. People get tempted to use them in weird ways if they are and then we're tied to a specific implementation (cause there's only one implementation at the moment so why not just pass the concrete type instead of the interface right? ;) ). If you need to allow setting them use a constructor function

Reply:
I'm overloading common here already.  Deferring this to a later cleanup step.